Attempt to update demo.sh for the mapping-tests

MakisH opened the pull request:
The [contrib/demo.sh](https://github.com/precice/aste/blob/mapping-tests/contrib/demo.sh) is currently out-of-date with the mapping-tests branch, leading to:

```shell
$ join_mesh.py mapped -i rbc.vtk -o result.vtk
usage: join_mesh.py [-h] [--out OUT_MESHNAME] [-r RECOVERY] [--numparts NUMPARTS] [--log {DEBUG,INFO,WARNING,ERROR,CRITICAL}] inputmesh
join_mesh.py: error: unrecognized arguments: -i rbc.vtk
```

I do not really understand how to modify it. I understand that it should be something like:

```shell
join_mesh.py -o result.vtk -r colored.dt0 -n 2 <output of preciceMapping>
```

I made it work with `rbc.vtk` as `<output of preciceMapping>`, but this is not correct. 

<details><summary>I tried with `mapped` (the directory), but it  also complains: (click)</summary>

```shell
$ join_mesh.py -o result.vtk -r colored.dt0 -n 2 mapped
INFO:root:Starting mesh recovery
INFO:root:Assembling recovered mesh
Traceback (most recent call last):
  File "/home/makish/github/aste/build/join_mesh.py", line 126, in <module>
    main()
  File "/home/makish/github/aste/build/join_mesh.py", line 16, in main
    mesh = read_mesh(args.in_meshname, args.numparts, args.recovery)
  File "/home/makish/github/aste/build/join_mesh.py", line 109, in read_mesh
    return join_mesh_recovery(dirname, partitions, recoveryPath)
  File "/home/makish/github/aste/build/join_mesh.py", line 88, in join_mesh_recovery
    return mesh_io.Mesh(all_points, all_cells, all_values)
  File "/home/makish/github/aste/build/mesh_io.py", line 29, in __init__
    self.cells = cells
  File "/home/makish/github/aste/build/mesh_io.py", line 89, in cells
    cellsbysize = {
  File "/home/makish/github/aste/build/mesh_io.py", line 89, in <dictcomp>
    cellsbysize = {
  File "/home/makish/github/aste/build/mesh_io.py", line 91, in <lambda>
    for k, v in itertools.groupby(cells, lambda cell: len(cell))
TypeError: object of type 'float' has no len()
```
</details>

@fsimonis any idea?

(see also https://github.com/precice/aste/issues/16)

fsimonis commented:
You found a bug. I mixed up 2 positional arguments.

Fixed in 9fe2cba, you can rebase this PR and it should  work.

>  I do not really understand how to modify it. I understand that it should be something like:
> join_mesh.py -o result.vtk -r colored.dt0 -n 2 <output of preciceMapping>

Yes exactly.

Explicitly passing the partition count should be optional. However, it looks like is not implemented yet.

MakisH commented:
@fsimonis I corrected the arguments the `demo.sh`, I added a call to `eval_mesh.py -d`, and I documented the respective unclear flags in the README. Please check if the changes make sense.

We definitely need more documentation, but let's treat it in a separate branch (preferably after the `mapping-tests` gets merged to `develop`). I opened #19 .
